src/plugins/intel_cpu/tests/unit/graph/inplace_resolve_io.cpp

```diff
@@ -0,0 +1,285 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <gtest/gtest.h>
+
+#include "dummy_node.hpp"
+#include "graph.h"
+
+#include "nodes/reorder.h"
+#include "nodes/input.h"
+#include "nodes/transpose.h"
+
+#include "openvino/op/result.hpp"
+#include "openvino/op/parameter.hpp"
+
+#include "common_test_utils/node_builders/constant.hpp"
+#include "common_test_utils/node_builders/eltwise.hpp"
+#include "common_test_utils/ov_tensor_utils.hpp"
+
+using namespace ov::intel_cpu;
+using namespace ov::op;
+
+// helper to check the inplace direction of a node with a part of its name
+void CheckInplaceDirection(const std::shared_ptr<Graph> graph, const std::string &patial_name, size_t inport, const Edge::LOOK undesiredDirection) {
```

Comment:
Isn't it better to check for "desired" direction?

Reply:
There should be a set of "desired" directions, v.s. only one "undesired".

Reply:
To my understanding the may be only two either UP or DOWN. So obviously only one of them is desired and one undesired, isn't it?

Reply:
There could be others - NONE. Some cases both NONE and DOWN are desiring. Some cases both NONE and UP are desiring.